In [1]:
import pandas as pd
import time
import matplotlib.pyplot as plt
import numpy as np

In [2]:
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)

In [3]:
shotData = pd.read_csv(r'2020_21_EPL_Shot.csv')

In [4]:
shotData.head()

h_team      id  minute       result      X      Y        xG         player h_a  player_id situation  year   shotType  match_id            a_team  h_goals  a_goals                 date   player_assisted lastAction
0  Arsenal  399642      21  MissedShots  0.719  0.404  0.014850  Thomas Partey   h       2328  OpenPlay  2020  RightFoot     14615  Newcastle United        3        0  2021-01-18 20:00:00       Bukayo Saka       Pass
1  Arsenal  399654      62  BlockedShot  0.689  0.507  0.015112   Granit Xhaka   h        204  OpenPlay  2020   LeftFoot     14615  Newcastle United        3        0  2021-01-18 20:00:00  Emile Smith-Rowe       Pass
2  Arsenal  393084      91   ShotOnPost  0.850  0.489  0.024479    Rob Holding   h       1749  SetPiece  2020       Head     14555       Southampton        1        1  2020-12-16 18:00:00       Bukayo Saka      Cross
3  Arsenal  377757      67  BlockedShot  0.778  0.437  0.030592   Jarrod Bowen   a       1776  OpenPlay  2020   LeftFoot     14099          West Ham        2        1  2020-09-19 19:00:00               NaN       None
4  Arsenal  381200      27    SavedShot  0.891  0.564  0.055482  Eddie Nketiah   h       6482  OpenPlay  2020       Head     14465  Sheffield United        2        1  2020-10-04 13:00:00    Kieran Tierney      Cross

In [5]:
#creating a data frame for Chelsea's Home matches
chelsea_home = shotData[shotData['h_team'] == 'Chelsea'].reset_index()
chelsea_home.head()

index   h_team      id  minute       result      X      Y        xG           player h_a  player_id       situation  year   shotType  match_id                a_team  h_goals  a_goals                 date     player_assisted    lastAction
0   1855  Chelsea  412201      55  MissedShots  0.769  0.799  0.034582      Mason Mount   h       7768  DirectFreekick  2020  RightFoot     14727  West Bromwich Albion        2        5  2021-04-03 11:30:00                 NaN      Standard
1   1856  Chelsea  405113      23  BlockedShot  0.807  0.509  0.051478    Mateo Kovacic   h       2254        OpenPlay  2020  RightFoot     14667      Newcastle United        2        0  2021-02-15 20:00:00  Callum Hudson-Odoi          Pass
2   1857  Chelsea  390461      56  MissedShots  0.879  0.437  0.081568         Raphinha   a       8026      FromCorner  2020   LeftFoot     14538                 Leeds        3        1  2020-12-05 20:00:00                 NaN       Rebound
3   1858  Chelsea  412199      50  BlockedShot  0.744  0.390  0.025456  Matheus Pereira   a       7153        OpenPlay  2020   LeftFoot     14727  West Bromwich Albion        2        5  2021-04-03 11:30:00     Callum Robinson  BallRecovery
4   1859  Chelsea  405114      25  MissedShots  0.927  0.554  0.441421      Timo Werner   h         65        OpenPlay  2020  RightFoot     14667      Newcastle United        2        0  2021-02-15 20:00:00       Marcos Alonso      HeadPass

In [6]:
#creating a data frame for Chelsea's Home match against leeds
chelsea_leeds = chelsea_home[chelsea_home['a_team'] == 'Leeds']
chelsea_leeds

index   h_team      id  minute       result      X      Y        xG             player h_a  player_id   situation  year   shotType  match_id a_team  h_goals  a_goals                 date  player_assisted    lastAction
2     1857  Chelsea  390461      56  MissedShots  0.879  0.437  0.081568           Raphinha   a       8026  FromCorner  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00              NaN       Rebound
46    1901  Chelsea  390471      77    SavedShot  0.811  0.501  0.051699  Ian Poveda-Ocampo   a       8723    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00              NaN        TakeOn
86    1941  Chelsea  390460      56  BlockedShot  0.843  0.474  0.028152           Raphinha   a       8026  FromCorner  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00    Mateusz Klich       Chipped
103   1958  Chelsea  390457      41    SavedShot  0.910  0.565  0.018462       Ben Chilwell   h        782  FromCorner  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00      Mason Mount        Aerial
144   1999  Chelsea  390470      76  MissedShots  0.922  0.463  0.306361     Olivier Giroud   h        502  FromCorner  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00      Mason Mount         Cross
148   2003  Chelsea  390454      22  MissedShots  0.759  0.437  0.030104      Stuart Dallas   a       8718    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00    Mateusz Klich          Pass
162   2017  Chelsea  390459      53  MissedShots  0.974  0.554  0.566574        Kai Havertz   h       5220    SetPiece  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00      Reece James         Cross
170   2025  Chelsea  390451      12  MissedShots  0.919  0.474  0.149094      Jack Harrison   a       8720    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00         Raphinha          Pass
173   2028  Chelsea  390467      67  MissedShots  0.951  0.557  0.591652     Olivier Giroud   h        502    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00              NaN          None
179   2034  Chelsea  390456      26         Goal  0.948  0.441  0.461563     Olivier Giroud   h        502    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00      Reece James         Cross
195   2050  Chelsea  390449       8  BlockedShot  0.937  0.575  0.082173     Olivier Giroud   h        502  FromCorner  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00      Mason Mount         Cross
217   2072  Chelsea  390462      59    SavedShot  0.898  0.410  0.116803        Timo Werner   h         65    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00   Olivier Giroud          Pass
240   2095  Chelsea  390464      59  BlockedShot  0.961  0.436  0.156337        Mason Mount   h       7768    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00              NaN       Rebound
243   2098  Chelsea  390445       0    SavedShot  0.926  0.666  0.065082       Hakim Ziyech   h       8992    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00     Thiago Silva       Chipped
255   2110  Chelsea  390469      75    SavedShot  0.918  0.415  0.383267        Timo Werner   h         65    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00      Mason Mount          Pass
279   2134  Chelsea  390468      70    SavedShot  0.741  0.643  0.016950        Mason Mount   h       7768    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00    Mateo Kovacic          Pass
280   2135  Chelsea  390455      25    SavedShot  0.813  0.494  0.074730    Patrick Bamford   a        822    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00         Raphinha      

In [7]:
chelsea_leeds['player'].value_counts().sort_values(ascending = False)

Olivier Giroud       6
Timo Werner          5
Christian Pulisic    4
Patrick Bamford      3
Raphinha             2
Mason Mount          2
Ian Poveda-Ocampo    1
Ben Chilwell         1
Stuart Dallas        1
Kai Havertz          1
Jack Harrison        1
Hakim Ziyech         1
Tammy Abraham        1
Reece James          1
Kurt Zouma           1
Name: player, dtype: int64

- Olivier Giroud had the most shots(6) for both teams in this match
- Next on the list is Timo Werner with 5 shots who is also a chelsea player
- The Leeds player with the most shots in this match was Patrick Bamford with 3 

In [8]:
chelsea_leeds['player_assisted'].value_counts().sort_values(ascending = False)

Mason Mount        7
Mateusz Klich      3
Olivier Giroud     3
Timo Werner        3
Reece James        2
Raphinha           2
Thiago Silva       1
Mateo Kovacic      1
Ben Chilwell       1
Kalvin Phillips    1
Kai Havertz        1
Name: player_assisted, dtype: int64

- Mason Mount created the most chances in this match
- Mateusz Klich was the Leeds player with the most chances created

In [9]:
chelsea_leeds['player'].value_counts().sum()

31

Total number of shots taken by both teams in this match was 31

In [10]:
# Which player had the highest number of expected goals?

chelsea_leeds.groupby('player').xG.sum().sort_values(ascending = False)

player
Olivier Giroud       1.567408
Timo Werner          1.506100
Christian Pulisic    0.900202
Patrick Bamford      0.689376
Kai Havertz          0.566574
Mason Mount          0.173287
Jack Harrison        0.149094
Raphinha             0.109720
Kurt Zouma           0.101091
Tammy Abraham        0.067006
Hakim Ziyech         0.065082
Ian Poveda-Ocampo    0.051699
Reece James          0.044397
Stuart Dallas        0.030104
Ben Chilwell         0.018462
Name: xG, dtype: float64

In [11]:
# Shot data for OLivier Giroud against Leeds
Giroud = chelsea_leeds[chelsea_leeds['player_id']== 502]
Giroud

index   h_team      id  minute       result      X      Y        xG          player h_a  player_id   situation  year   shotType  match_id a_team  h_goals  a_goals                 date player_assisted    lastAction
144   1999  Chelsea  390470      76  MissedShots  0.922  0.463  0.306361  Olivier Giroud   h        502  FromCorner  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00     Mason Mount         Cross
173   2028  Chelsea  390467      67  MissedShots  0.951  0.557  0.591652  Olivier Giroud   h        502    OpenPlay  2020  RightFoot     14538  Leeds        3        1  2020-12-05 20:00:00             NaN          None
179   2034  Chelsea  390456      26         Goal  0.948  0.441  0.461563  Olivier Giroud   h        502    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00     Reece James         Cross
195   2050  Chelsea  390449       8  BlockedShot  0.937  0.575  0.082173  Olivier Giroud   h        502  FromCorner  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00     Mason Mount         Cross
290   2145  Chelsea  390453      21  MissedShots  0.817  0.522  0.046958  Olivier Giroud   h        502    OpenPlay  2020   LeftFoot     14538  Leeds        3        1  2020-12-05 20:00:00             NaN  BallRecovery
328   2183  Chelsea  390446       1  MissedShots  0.956  0.587  0.078701  Olivier Giroud   h        502  FromCorner  2020       Head     14538  Leeds        3        1  2020-12-05 20:00:00     Mason Mount         Cross

In [12]:
tableData = shotData.drop(['minute', 'result', 'X', 'Y', 'player', 'situation', 'year', 'shotType', 'player_assisted', 'lastAction'], axis=1)
tableData.head()

h_team      id        xG h_a  player_id  match_id            a_team  h_goals  a_goals                 date
0  Arsenal  399642  0.014850   h       2328     14615  Newcastle United        3        0  2021-01-18 20:00:00
1  Arsenal  399654  0.015112   h        204     14615  Newcastle United        3        0  2021-01-18 20:00:00
2  Arsenal  393084  0.024479   h       1749     14555       Southampton        1        1  2020-12-16 18:00:00
3  Arsenal  377757  0.030592   a       1776     14099          West Ham        2        1  2020-09-19 19:00:00
4  Arsenal  381200  0.055482   h       6482     14465  Sheffield United        2        1  2020-10-04 13:00:00

In [13]:
len(tableData)

9230

In [14]:
tableData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9230 entries, 0 to 9229
Data columns (total 10 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   h_team     9230 non-null   object 
 1   id         9230 non-null   int64  
 2   xG         9230 non-null   float64
 3   h_a        9230 non-null   object 
 4   player_id  9230 non-null   int64  
 5   match_id   9230 non-null   int64  
 6   a_team     9230 non-null   object 
 7   h_goals    9230 non-null   int64  
 8   a_goals    9230 non-null   int64  
 9   date       9230 non-null   object 
dtypes: float64(1), int64(5), object(4)
memory usage: 576.9+ KB


In [15]:
tableData.columns

Index(['h_team', 'id', 'xG', 'h_a', 'player_id', 'match_id', 'a_team', 'h_goals', 'a_goals', 'date'], dtype='object')

In [16]:
matchId = tableData['match_id'].unique()
len(matchId)

380

There 380 unique match Id's in this data set which corresponds with the total number of matches in a season. The data for all matches in a single season is complete.

In [17]:
tableData.groupby('h_team').match_id.nunique()

h_team
Arsenal                    19
Aston Villa                19
Brighton                   19
Burnley                    19
Chelsea                    19
Crystal Palace             19
Everton                    19
Fulham                     19
Leeds                      19
Leicester                  19
Liverpool                  19
Manchester City            19
Manchester United          19
Newcastle United           19
Sheffield United           19
Southampton                19
Tottenham                  19
West Bromwich Albion       19
West Ham                   19
Wolverhampton Wanderers    19
Name: match_id, dtype: int64

- There are 38 game weeks in  season
- There are 20 teams in the league
- Teams play against each other twice in a season i.e 19 x 2 times = 38 game weeks
- Teams plays against each other which results in 10 matches per game week
- Total number of matches i.e 38 x 10 = 380 matches

In [18]:

homeTeam = tableData['h_team'].unique()
homeTeam

array(['Arsenal', 'Aston Villa', 'Brighton', 'Burnley', 'Chelsea',
       'Crystal Palace', 'Everton', 'Fulham', 'Leeds', 'Leicester',
       'Liverpool', 'Manchester City', 'Manchester United',
       'Newcastle United', 'Sheffield United', 'Southampton', 'Tottenham',
       'West Bromwich Albion', 'West Ham', 'Wolverhampton Wanderers'],
      dtype=object)

In [42]:
awayTeam = tableData['a_team'].unique()
awayTeam

array(['Newcastle United', 'Southampton', 'West Ham', 'Sheffield United',
       'Leeds', 'Everton', 'Aston Villa', 'Fulham', 'Tottenham',
       'Burnley', 'Wolverhampton Wanderers', 'Manchester United',
       'Chelsea', 'West Bromwich Albion', 'Liverpool', 'Brighton',
       'Leicester', 'Manchester City', 'Crystal Palace', 'Arsenal'],
      dtype=object)

In [19]:
tableData['Date'] = tableData['date'].str.slice(start=0, stop=10)
tableData['Date'] = pd.to_datetime(tableData['Date'])
tableData['Date']

0      2021-01-18
1      2021-01-18
2      2020-12-16
3      2020-09-19
4      2020-10-04
          ...    
9225   2020-12-12
9226   2021-04-17
9227   2021-05-09
9228   2021-05-09
9229   2020-10-04
Name: Date, Length: 9230, dtype: datetime64[ns]

In [20]:
tableData['Date'].nunique()

135

All 380 matches were played in 135  match days

In [21]:
#tableData['Time'] = tableData['date'].str.slice(start=11, stop=19)
tableData['Time'] =  pd.to_datetime(tableData['date'])
tableData['Time'] = tableData['Time'].apply(lambda x:x.time())
tableData['Time']

0       20:00:00
1       20:00:00
2       18:00:00
3       19:00:00
4       13:00:00
          ...   
9225    12:30:00
9226    19:15:00
9227    11:00:00
9228    11:00:00
9229    13:00:00
Name: Time, Length: 9230, dtype: object

In [22]:
tableData.drop(['date'], axis=1)

h_team      id        xG h_a  player_id  match_id            a_team  h_goals  a_goals       Date      Time
0                     Arsenal  399642  0.014850   h       2328     14615  Newcastle United        3        0 2021-01-18  20:00:00
1                     Arsenal  399654  0.015112   h        204     14615  Newcastle United        3        0 2021-01-18  20:00:00
2                     Arsenal  393084  0.024479   h       1749     14555       Southampton        1        1 2020-12-16  18:00:00
3                     Arsenal  377757  0.030592   a       1776     14099          West Ham        2        1 2020-09-19  19:00:00
4                     Arsenal  381200  0.055482   h       6482     14465  Sheffield United        2        1 2020-10-04  13:00:00
...                       ...     ...       ...  ..        ...       ...               ...      ...      ...        ...       ...
9225  Wolverhampton Wanderers  391403  0.089113   h       8291     14552       Aston Villa        0        1 2020-12-12  12:30:00
9226  Wolverhampton Wanderers  415501  0.054968   a       7707     14754  Sheffield United        1        0 2021-04-17  19:15:00
9227  Wolverhampton Wanderers  420442  0.048403   h        900     14784          Brighton        2        1 2021-05-09  11:00:00
9228  Wolverhampton Wanderers  420429  0.014879   h       6853     14784          Brighton        2        1 2021-05-09  11:00:00
9229  Wolverhampton Wanderers  381218  0.046761   a        773     14474            Fulham        1        0 2020-10-04  13:00:00

[9230 rows x 11 columns]

In [23]:
teams = pd.DataFrame(tableData.groupby([ 'match_id','h_team','a_team','h_a','Date','Time', 'h_goals', 'a_goals']).match_id.unique())
tableData['h_goals'] = pd.to_numeric(tableData['h_goals'])
tableData['a_goals'] = pd.to_numeric(tableData['a_goals'])
teams.sort_values(by='Date' , ascending=True).head()


match_id
match_id h_team    a_team           h_a Date       Time     h_goals a_goals         
14086    Fulham    Arsenal          a   2020-09-12 11:30:00 0       3        [14086]
14091    West Ham  Newcastle United h   2020-09-12 19:00:00 0       2        [14091]
14090    Liverpool Leeds            h   2020-09-12 16:30:00 4       3        [14090]
                                    a   2020-09-12 16:30:00 4       3        [14090]
14091    West Ham  Newcastle United a   2020-09-12 19:00:00 0       2        [14091]

In [24]:
teams['match_id'].unique

<bound method Series.unique of match_id  h_team                   a_team             h_a  Date        Time      h_goals  a_goals
14086     Fulham                   Arsenal            a    2020-09-12  11:30:00  0        3          [14086]
                                                      h    2020-09-12  11:30:00  0        3          [14086]
14087     Crystal Palace           Southampton        a    2020-09-12  14:00:00  1        0          [14087]
                                                      h    2020-09-12  14:00:00  1        0          [14087]
14088     Burnley                  Manchester United  a    2021-01-12  22:15:00  0        1          [14088]
                                                                                                      ...   
14812     Sheffield United         Burnley            h    2021-05-23  15:00:00  1        0          [14812]
14813     West Ham                 Southampton        a    2021-05-23  15:00:00  3        0          [14813]

In [25]:
teams.describe()

match_id
count       760
unique      380
top     [14086]
freq          2

In [26]:
teams.head()

match_id
match_id h_team         a_team            h_a Date       Time     h_goals a_goals         
14086    Fulham         Arsenal           a   2020-09-12 11:30:00 0       3        [14086]
                                          h   2020-09-12 11:30:00 0       3        [14086]
14087    Crystal Palace Southampton       a   2020-09-12 14:00:00 1       0        [14087]
                                          h   2020-09-12 14:00:00 1       0        [14087]
14088    Burnley        Manchester United a   2021-01-12 22:15:00 0       1        [14088]

In [27]:
tableData.head()

h_team      id        xG h_a  player_id  match_id            a_team  h_goals  a_goals                 date       Date      Time
0  Arsenal  399642  0.014850   h       2328     14615  Newcastle United        3        0  2021-01-18 20:00:00 2021-01-18  20:00:00
1  Arsenal  399654  0.015112   h        204     14615  Newcastle United        3        0  2021-01-18 20:00:00 2021-01-18  20:00:00
2  Arsenal  393084  0.024479   h       1749     14555       Southampton        1        1  2020-12-16 18:00:00 2020-12-16  18:00:00
3  Arsenal  377757  0.030592   a       1776     14099          West Ham        2        1  2020-09-19 19:00:00 2020-09-19  19:00:00
4  Arsenal  381200  0.055482   h       6482     14465  Sheffield United        2        1  2020-10-04 13:00:00 2020-10-04  13:00:00

In [28]:
tableData.columns

Index(['h_team', 'id', 'xG', 'h_a', 'player_id', 'match_id', 'a_team', 'h_goals', 'a_goals', 'date', 'Date', 'Time'], dtype='object')

In [29]:
results = tableData.drop(['id', 'xG', 'player_id','date'], axis = 1)

In [30]:
# results = results.rename(columns={'h_team':'Home','a_team':'Away','h_goals':'Goals(H)','a_goals':'Goals(A)'}, inplace=True) 

In [31]:
results.tail()

h_team h_a  match_id            a_team  h_goals  a_goals       Date      Time
9225  Wolverhampton Wanderers   h     14552       Aston Villa        0        1 2020-12-12  12:30:00
9226  Wolverhampton Wanderers   a     14754  Sheffield United        1        0 2021-04-17  19:15:00
9227  Wolverhampton Wanderers   h     14784          Brighton        2        1 2021-05-09  11:00:00
9228  Wolverhampton Wanderers   h     14784          Brighton        2        1 2021-05-09  11:00:00
9229  Wolverhampton Wanderers   a     14474            Fulham        1        0 2020-10-04  13:00:00

In [85]:
score = results['h_goals']
score_opp = results['a_goals']

In [86]:
points_map = {
    'W': 3,
    'D': 1,
    'L': 0
}

def get_result(score, score_opp):
    if score == score_opp:
        return 'D'
    elif score > score_opp:
        return 'W'
    else:
        return 'L'

In [87]:
# full time goals
results['Goals'] = np.where(results['h_team'] != results['a_team'],
                                 results['h_goals'],
                                 results['a_goals'])
results['Goals_Opp'] = np.where(results['a_team'] != results['h_team'],
                                     results['a_goals'],
                                     results['h_goals'])
results['Result'] = np.vectorize(get_result)(results['Goals'], results['Goals_Opp'])
results['Points'] = results['Result'].map(points_map)


In [88]:
team_results = (results
.sort_values(by=['Date', 'Time']))

In [89]:
team_results.head()

h_team h_a  match_id   a_team  h_goals  a_goals       Date      Time  Goals  Goals_Opp Result  Points
3232  Fulham   a     14086  Arsenal        0        3 2020-09-12  11:30:00      0          3      L       0
3239  Fulham   a     14086  Arsenal        0        3 2020-09-12  11:30:00      0          3      L       0
3275  Fulham   h     14086  Arsenal        0        3 2020-09-12  11:30:00      0          3      L       0
3307  Fulham   h     14086  Arsenal        0        3 2020-09-12  11:30:00      0          3      L       0
3309  Fulham   a     14086  Arsenal        0        3 2020-09-12  11:30:00      0          3      L       0

In [90]:
team_results = team_results.groupby(['match_id', 'Date', 'Time', 'h_team', 'a_team'])['Goals','Goals_Opp','Points'].mean()

C:\Users\SOLA\AppData\Local\Temp/ipykernel_8536/1188176156.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  team_results = team_results.groupby(['match_id', 'Date', 'Time', 'h_team', 'a_team'])['Goals','Goals_Opp','Points'].mean()


## Calculating Home Results League Table (Home Standings Table)

In [91]:
# Testing the standings function against real world data
(team_results
     .groupby('h_team')
     .sum()['Points']
     .sort_values(ascending=False))

h_team
Manchester City            41.0
West Ham                   34.0
Liverpool                  33.0
Tottenham                  33.0
Chelsea                    33.0
Manchester United          31.0
Leeds                      29.0
Leicester                  28.0
Arsenal                    28.0
Southampton                27.0
Aston Villa                25.0
Wolverhampton Wanderers    25.0
Crystal Palace             23.0
Newcastle United           23.0
Everton                    22.0
Brighton                   21.0
Burnley                    18.0
Sheffield United           16.0
West Bromwich Albion       15.0
Fulham                     10.0
Name: Points, dtype: float64